src/extension/sdk/dev_tools/manager.ts

```diff
@@ -325,13 +325,13 @@ export class DevToolsManager implements vs.Disposable {
 		// to only use that for a title, we may be able to relax that.
 		if (options.location !== "external") {
 			queryParams.embed = "true";
-			const fullUrl = await this.buildDevToolsUrl(url, queryParams, vmServiceUri);
+			const fullUrl = await this.buildDevToolsUrl(url, queryParams, vmServiceUri, session?.clientVmServiceUri);
```

Comment:
I would pull this out to a variable (and reuse below) with the same condition as `vmServiceUri` on line 323 so that we don't need to worry about `buildDevToolsUrl` accidentally using it for static tools.

We could actually flatten the implementation in `buildDevToolsUrl` a little then too (removing the outer `if` statement):

```
if (clientVmServiceUri) {
	queryParams.uri = clientVmServiceUri;
} else if (vmServiceUri) {
	const exposedUrl = await envUtils.exposeUrl(vmServiceUri, this.logger);
	queryParams.uri = exposedUrl;
}
```

Reply:
I ended up not using the additional client URI because I had them backwards - the VM service URI is already accessible on mac, but asking for `exposeUrl` makes it only accessible on linux.

Reply:
Hmmm, this sounds a little unsound to me. There are many places that we use this VM Service URI, and only DevTools client is running on the Mac. So if the VM Service URI we get is the raw Mac one (and not accessible from Linux where the extension host is), we may pass it to things on Linux that won't be able to connect (for example we may [give it to DevTools server](https://github.com/Dart-Code/Dart-Code/blob/7a0efbc0ade74e3d617e43b483b96a143705fea4/src/extension/sdk/dev_tools/manager.ts#L464)).

I feel like it may be better if we still had access to both URIs in Dart-Code so we can use the appropriate URI for each use case?